app/assets/javascripts/samples/sample_datatable.js

```diff
@@ -971,12 +999,200 @@ function changeToEditMode() {
     });
   }
 
+  function initEditColumns() {
+    function cancelEditMode() {
+      dropdownList.find('.text-edit').hide();
+      dropdownList.find('.controls .ok,.cancel').hide();
+      dropdownList.find('.text').css('display', ''); // show() doesn't work
+      dropdownList.find('.controls .vis,.edit,.del').css('display', ''); // show() doesn't work
+      columnEditMode = false;
+    }
+
+    function editColumn(li) {
+      var id = li.attr('data-id');
+      var text = li.find('.text');
+      var textEdit = li.find('.text-edit');
+      var newName = textEdit.val().trim();
+      var url = li.attr('data-edit-url');
+
+      $.ajax({
+        url: url,
+        type: 'PUT',
+        data: {custom_field: {name: newName}},
+        dataType: 'json',
+        success: function() {
+          text.text(newName);
+          $(table.columns().header()).filter('#' + id).text(newName);
+          cancelEditMode();
+        },
+        error: function(xhr) {
+          // TODO
+        }
+      });
+    }
+
+    // On edit buttons click (bind onto master dropdown list)
+    dropdownList.on('click', '.edit:not(.disabled)', function(event) {
+      event.stopPropagation();
+
+      cancelEditMode();
+
+      var self = $(this);
+      var li = self.closest('li');
+      var text = li.find('.text');
+      var textEdit = li.find('.text-edit');
+      var controls = li.find('.controls .vis,.edit,.del');
+      var controlsEdit = li.find('.controls .ok,.cancel');
+
+      // Toggle edit mode
+      columnEditMode = true;
+      li.addClass('editing');
+
+      // Set the text-edit's value
+      textEdit.val(text.text().trim());
+
+      // Toggle elements
+      text.hide();
+      controls.hide();
+      textEdit.css('display', ''); // show() doesn't work
+      controlsEdit.css('display', ''); // show() doesn't work
+
+      // Focus input
+      textEdit.focus();
+    });
+
+    // On hiding dropdown, cancel edit mode throughout dropdown
+    dropdown.on('hidden.bs.dropdown', function() {
+      cancelEditMode();
+    });
+
+    // On ok buttons click
+    dropdownList.on('click', '.ok', function(event) {
+      event.stopPropagation();
+      var self = $(this);
+      var li = self.closest('li');
+      editColumn(li);
+    });
+
+    // On enter click while editing column text
+    dropdownList.on('keydown', 'input.text-edit', function(event) {
+      if (event.keyCode === 13) {
+        event.preventDefault();
```

Comment:
Maybe use `event.stopPropagation();` instead, as in everywhere else? If it doesn't make a difference of course.